warehouse/multitenant/manager.go

```diff
@@ -0,0 +1,139 @@
+package multitenant
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/rudderlabs/rudder-server/config"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+)
+
+var degradedWorkspaceIDs []string
+
+func init() {
+	config.RegisterStringSliceConfigVariable(nil, &degradedWorkspaceIDs, false, "Warehouse.degradedWorkspaceIDs")
+}
+
+type Manager struct {
+	BackendConfig        backendconfig.BackendConfig
+	DegradedWorkspaceIDs []string
+
+	sourceIDToWorkspaceID map[string]string
+	excludeWorkspaceIDMap map[string]struct{}
+
+	ready     chan struct{}
+	sourceMu  sync.Mutex
+	readyOnce sync.Once
+	initOnce  sync.Once
+}
+
+func (m *Manager) init() {
+	m.initOnce.Do(func() {
+		if m.DegradedWorkspaceIDs == nil {
+			m.DegradedWorkspaceIDs = degradedWorkspaceIDs
+		}
+
+		m.sourceIDToWorkspaceID = make(map[string]string)
+		m.excludeWorkspaceIDMap = make(map[string]struct{})
```

Comment:
We are not using this anywhere?

Reply:
I believe we are `degradedWorkspaceIDs` as `excludeWorkspace`. Do we still need this.